pypredict/trades.py

```diff
@@ -98,18 +105,63 @@ def run(self):
 
         asyncio.get_event_loop().run_until_complete(self.subscribe())
 
+    def build_model(self):
+        model = compose.Pipeline(
+            ('scale', preprocessing.StandardScaler()),
+            ('lin_reg', linear_model.LinearRegression())
+        )
+        return model
+    
+    def get_timestamp(self, epoch):
+        """
+        converts unix epoch to datetime
+        """
+        epoch_time = epoch / 1000.0
+        timestamp = datetime.fromtimestamp(epoch_time)
+        return timestamp
+    
+    async def run_model_pipeline(self, data):
+        """
+        Run an online model using river.
+        The model will first make a prediction and then
+        continuously learn as it gets new price data.
+        """
+        # convert unix epoch to datetime
+        timestamp = self.get_timestamp(data["timestamp"])
+        # extract the microsecond component and use it as a model feature
+        x = {"microsecond" : timestamp.microsecond}
+        # generate a prediction
+        price_pred = round(self.model.predict_one(x), 4)
+        price = data["price"]
+        print(timestamp, price, price_pred)
+        # pass the actual trade price to the model
+        self.model.learn_one(x, price)
```

Comment:
So this is the part of the code that's CPU-bound? Do you have an idea of how long it takes? The one thing I'm curious about is since PyEnsign is built on coroutines, computing the model will block the subscribe and the subscriber might start to fall behind (although that might not be a big deal for this demo). If PyEnsign used threads and we were able to do the model computation in a separate thread then it would be less of an issue. Something to think about but this PR is mostly about the model pipeline so I think it's ok for now.

Reply:
Because the model is learning incrementally, the training and scoring is very fast - but the reality is that AAPL is one of the most actively traded stocks and so the predictions are likely getting computed after the trade has executed (and there is already the latency from us getting this from an external API and publishing to a topic).  You are correct that it's less of an issue if we ran the model pipeline in a separate thread, though.  The reality is that this example is somewhat contrived ;-).